src/reactionsystem.jl

```diff
@@ -0,0 +1,152 @@
+    convert(ODESystem, rs, defaults=defaults)
+end
+
+""" ODESystem constructor """
+function ModelingToolkit.ODESystem(sbmlfile::String; kwargs...)
+    model = readSBML(sbmlfile)
+    ODESystem(model; kwargs...)
+end
+
+""" ODEProblem constructor """
+function ModelingToolkit.ODEProblem(model::Model,tspan;kwargs...)  # PL: Todo: add u0 and parameters argument
+    odesys = ODESystem(model,kwargs...)
+    ODEProblem(odesys, [], tspan)
+end
+
+""" ODEProblem constructor """
+function ModelingToolkit.ODEProblem(sbmlfile::String,tspan;kwargs...)  # PL: Todo: add u0 and parameters argument
+    odesys = ODESystem(sbmlfile;kwargs...)
+    ODEProblem(odesys, [], tspan)
+end
+
+""" Convert intensive to extensive expressions """
+function make_extensive(model)
+    model = to_initial_amounts(model)
+    model = to_extensive_math(model)
+    model  # Todo: For spevies with `hOSU=false` multiply all occurences in mathematical expressions by compartment size.
+           # Also convert species initialConcentrations to initialAmounts
+end
+
+""" Convert initial_concentration to initial_amount """
+function to_initial_amounts(model::Model)
+    model = deepcopy(model)
+    for specie in model.species
+        if isequal(specie.initial_amount, nothing)
+            compartment = model.compartments[specie.compartment]
+            specie.initial_amount = specie.initial_concentration * compartment.size
+            specie.initial_concentration = nothing
+        end
+    end
+    model
+end
+
+""" Convert intensive to extensive mathematical expression """
+function to_extensive_math(model::Model)
+    model = deepcopy(model)
+    for reaction in model.reactions
+        km = reaction.kinetic_math
+        reaction.km = ...  # PL: Todo: @Anand can you multiply species with `hOSU=true` with their compartment volume?
+    end
+    reaction
+end
+
+""" Expand reversible reactions to two reactions """
+function expand_reversible(model)
+    model  # Todo: convert all Reactions that are `reversible=true` to a forward and reverse reaction with `reversible=false`.
+end
+
+""" Convert SBML.Reaction to MTK.Reaction """
+function mtk_reaction(reaction::SBML.Reaction)
+    reactants = []
+    rstoich = []
+    products = []
+    pstoich = []
+    for (k,v) in reaction.stoichiometry
+        if v < 0
+            push!(reactants, create_var(k))
+            push!(rstoich, -v)
+        elseif v > 0
+            push!(products, create_var(k))
+            push!(pstoich, -v)
+        else
+            @error("Stoichiometry of $k must be non-zero")
+        end
+    end
+    subsdict = _get_substitutions(model)
+    # PL: Todo: @Anand: can you convert kinetic_math to Symbolic expression. Perhaps it would actually better if kinetic Math would be a Symbolics.jl expression rather than of type `Math`? But Mirek wants `Math`, I think.
+    kl = substitute(reaction.kinetic_math, subsdict)  # PL: Todo: might need conversion of kinetic_math to Symbolic MTK expression
```

Comment:
IIRC @exaexa was taking this Math* sublanguage conversion. But I can get started on it for sure